### PR TITLE
WEBGUI: Comments, Formatting, Filter widget changes

### PR DESCRIPTION
**Pull request by bobszi**
Comments inserted for important parts of WEBGUI Javascript files.
Code formatting made more consistent.
Behaviour of Filter widget changed slightly(now we can delete any filter except the uppermost one.

### CODE REVIEW

File: viewer_clients/web-client/scripts/codecheckerviewer/BugStoreModelTree.js [hunk 1]
```diff
@@ -221,22 +226,47 @@ return declare(null, {
...
+
+
+      // that._queryReportsForFile(function(reports) {
+      //   var reportsComplete = 0;
+
+      //   reports.forEach(function(report) {
+      //     that._buildExecPathForReport(bugStoreDataTmp, report, function() {
+      //       ++reportsComplete;
+      //       if (reportsComplete === reports.length) {
+      //         // FIXME: it's slow on large array
```
Comment (dkrupp) [inline viewer_clients/web-client/scripts/codecheckerviewer/BugStoreModelTree.js]: remove this

File: viewer_clients/web-client/scripts/codecheckerviewer/CodeCheckerViewer.js [hunk 2]
```diff
@@ -295,15 +329,14 @@ return declare(null, {
...
       }
     });
 
     try {
-      /*newOverviewTC.overviewGrid.fillOverviewGrid(newOverviewTC.getStateOfFilters(),
-        newOverviewTC.overviewPager.getPagerParams());
-
-      newOverviewTC.overviewPager.disableArrowsAsNeeded();*/
+      /* newOverviewTC.overviewGrid.fillOverviewGrid(newOverviewTC.getStateOfFilters(),
+           newOverviewTC.overviewPager.getPagerParams());
```
Comment (dkrupp) [inline viewer_clients/web-client/scripts/codecheckerviewer/CodeCheckerViewer.js]: remove this

File: viewer_clients/web-client/scripts/codecheckerviewer/BugStoreModelTree.js [hunk 3]
```diff
@@ -221,22 +226,47 @@ return declare(null, {
     that._queryReportsForFile(function(reports) {
       var reportsComplete = 0;
 
-      reports.forEach(function(report) {
-        that._buildExecPathForReport(bugStoreDataTmp, report, function() {
+      for (var i = 0 , reportsLen = reports.length; i < reportsLen ; ++i) {
```
Comment (igalex) [inline viewer_clients/web-client/scripts/codecheckerviewer/BugStoreModelTree.js]: Keep forEach

File: viewer_clients/web-client/scripts/codecheckerviewer/BugStoreModelTree.js [hunk 4]
```diff
@@ -221,22 +226,47 @@ return declare(null, {
...
+        that._buildExecPathForReport(bugStoreDataTmp, reports[i], function() {
           ++reportsComplete;
           if (reportsComplete === reports.length) {
             // FIXME: it's slow on large array
-            bugStoreDataTmp.forEach(function(item) {
-              item.overwrite = true;
-              that.bugStore.put(item, item);
-            });
+            for (var j = 0 , bugStoreDataTmpLen = bugStoreDataTmp.length ; j < bugStoreDataTmpLen ; ++j) {
```
Comment (igalex) [inline viewer_clients/web-client/scripts/codecheckerviewer/BugStoreModelTree.js]: Keep forEach

**Pull request merged**